### PR TITLE
Agregar servicio restablecer contrasena

### PR DESCRIPTION
**Pull request by CamiloArias12**
# Services reset password

## Description
Se agregaron los servicos para restablecer la contrasena del usuario :

### Summary
1.Se conecto la pagina reset-password a la api 
image

2.Se agrego la pagina change-pasword para restablecer la contrasena del usuario
image

## How Has This Been Tested?
Se testeo directamente en la web

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/modules/auth/components/error-validate-password-reset/index.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export function ErrorValidatePasswordReset() {
+    useEffect(() => {
+
+        toast("Error al restablecer la contrasena intentalo nuevamente")
+    }, [])
+    const router = useRouter()
+    router.replace("/reset-password")
+
+
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/components/error-validate-password-reset/index.tsx:15]:
```suggestion
```

File: src/modules/auth/services/actions.ts [hunk 2]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+                message: 'Error'
+            }
+        }
+    }
+    return {
+        errors: validateFields.error.flatten().fieldErrors,
+        message: 'Error'
+    }
+
+}
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/actions.ts:65]: invertir if primero negativo y retornarlo

File: src/modules/auth/services/actions.ts [hunk 3]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+
+            }
+
+
+        }
+    }
+    return {
+        errors: validateFields.error?.flatten().fieldErrors,
+        message: 'Error'
+    }
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/actions.ts:102]: invertir los if primero negativo e invertir error de una asi no hacemos tanto nesting

File: src/modules/auth/services/index.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    try {
+        const response = await fetch(`${process.env.API_URL}/client/reset-password-token`,
+            {
+                method: 'POST',
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                body: JSON.stringify({ token: token, password: password })
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/index.ts:33]: se podria crear una funcion para apiPost y que reciba el tipo de dato ejemplo, `apiPost('/', {type: "application/json", data: {...}})`

File: src/modules/auth/templates/change-password.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import { FiArrowLeftCircle } from "react-icons/fi";
+import Link from "next/link";
+import { useFormState } from "react-dom";
+import { SubmitButton } from "@/modules/common/components/submit-button";
+import { changePassword } from "../services/actions";
+import { toast } from "react-toastify";
+import { useEffect } from "react";
+import { InputPassword } from "@/modules/common/components/input-password";
+import { ROUTES_AUTH } from "../types";
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/templates/change-password.tsx:11]:
```suggestion
```

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

File: src/modules/auth/components/error-validate-password-reset/index.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+"use client"
+import { useRouter } from "next/navigation"
+import { useEffect } from "react"
+import { toast } from "react-toastify"
+
+export function ErrorValidatePasswordReset() {
+    useEffect(() => {
+
+        toast("Error al restablecer la contrasena intentalo nuevamente")
```
Comment (JosephSC0121) [inline src/modules/auth/components/error-validate-password-reset/index.tsx:9]:
```suggestion
        toast("Error al restablecer la contraseña intentalo nuevamente")
```

File: src/modules/auth/components/error-validate-password-reset/index.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    }, [])
+    const router = useRouter()
+    router.replace("/reset-password")
+    return <></>
```
Comment (JosephSC0121) [inline src/modules/auth/components/error-validate-password-reset/index.tsx:13]: No puede ser confuso retornar nada? yo retornaria null, para saber si llegó a ese punto (solo pregunto)

File: src/modules/auth/services/actions.ts (hunk 2 above)
Comment (JosephSC0121) [reply src/modules/auth/services/actions.ts:65]: https://medium.com/swlh/return-early-pattern-3d18a41bba8
https://refactoring.guru/es/replace-nested-conditional-with-guard-clauses

File: src/modules/auth/services/actions.ts [hunk 8]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+        password: formData.get("password"),
+        passwordReply: formData.get("passwordReply")
+    })
+
+    if (validateFields.success) {
+        if (validateFields.data.password === validateFields.data.passwordReply) {
+            const response = await changePasswordService(validateFields.data.token, validateFields.data.password)
+            if (response) {
+                return {
+                    message: 'La contrasena ha sido actualizada'
```
Comment (JosephSC0121) [inline src/modules/auth/services/actions.ts:83]:
```suggestion
                    message: 'La contraseña ha sido actualizada'
```

File: src/modules/auth/components/error-validate-password-reset/index.tsx (hunk 7 above)
Comment (CamiloArias12) [reply src/modules/auth/components/error-validate-password-reset/index.tsx]: [ab52e02](https://github.com/OwlByTech/optitech-web/pull/6/commits/ab52e02bc6308542f0c67eb7b67502c0ffed75cc)

Comment (JosephSC0121) [reply src/modules/auth/components/error-validate-password-reset/index.tsx]: era este [ca84dac](https://github.com/OwlByTech/optitech-web/pull/6/commits/ca84dac461bd924a2f072ae16853189812c96e6e)

**Review by JosephSC0121 — APPROVED** [review]

**Pull request merged**